### PR TITLE
[Docs] Update `Quickstart` guide to onboard new devs

### PR DESCRIPTION
**Pull request by Olshansk**
## Summary

Update the Quickstar guide to enable anyone to start developing on LocalNet quickly.

## Issue
- #460

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have commented my code
- [ ] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

**Review by Olshansk — COMMENTED** [review]
@red-0ne @okdas Now looking for a deep review of the `Quickstart Guide`, but more of just an FYI.

PTAL at the auxiliary changes (configs, code, etc) in particular.

File: config.yml [hunk 1]
```diff
@@ -88,6 +88,10 @@ genesis:
     staking:
       params:
         bond_denom: upokt
+        # TODO_BLOCKER(@Olshansk): Figure out what this should be on Shannon
+        # re-genesis. We're setting it to 1 for Alpha TestNet #1 so Grove
+        # maintains the only validator until Alpha TestNet #2.
+        max_validators: 1
```
Comment (okdas) [inline config.yml:94]: Oh, this is great for the first public TestNet, actually! Let's figure out the validator power plan in the near future, though. :)

File: localnet/poktrolld/config/supplier1_stake_config.yaml [hunk 2]
```diff
@@ -15,5 +15,5 @@ services:
         rpc_type: JSON_RPC
   - service_id: ollama
     endpoints:
-      - publicly_exposed_url: http://relayminer1:8545
+      - publicly_exposed_url: http://relayminer2:8545
```
Comment (red-0ne) [inline localnet/poktrolld/config/supplier1_stake_config.yaml:18]: Not sure if this is intended to be a working (non-kubernetes) example, so maybe it should remain `relayminer1`

File: localnet/poktrolld/config/relayminer_config.yaml [hunk 3]
```diff
@@ -1,19 +1,19 @@
-signing_key_name: supplier1
+signing_key_name: shannon_supplier
```
Comment (red-0ne) [inline localnet/poktrolld/config/relayminer_config.yaml:1]: Make sure `shannon_supplier` signing key name exists in `config.yml` and that it has a stake target in `Makefile`

**Review by red-0ne — COMMENTED** [review]
Great guide, left a couple of comments but should not affect the quickstart story.

**Pull request merged**